### PR TITLE
Sync `transformers` and `accelerate` versions

### PR DESCRIPTION
**Pull request by michaelbenayoun**
# What does this PR do?

This PR synchronizes `optimum-neuron` with more recent `transformers` and `accelerate` versions:

- `accelerate==0.29.2`, which is the latest release when this PR is being done,
- `transformers==4.40.2`, which will be the latest releae when this PR is merged.

Related PR in `transformers`: https://github.com/huggingface/transformers/pull/30259

On top of that:

- The workflows for Trainium instances have been updated and use K8 now.

### CODE REVIEW

File: .github/workflows/test_trainium_common.yml
Comment (dacorvo) [inline .github/workflows/test_trainium_common.yml:22]:
```suggestion
```

File: optimum/neuron/trainers.py
Comment (dacorvo) [inline optimum/neuron/trainers.py:442]: I love when these lines pop magically to solve a sync issue ... 😜

**Review by dacorvo — COMMENTED** [review]
Awesome ! Looks good to me, but found a few nits !

**Comment by dacorvo** [comment]
You should unpin the safetensor package here because there is now a conflict:
https://github.com/huggingface/optimum-neuron/blob/1e7d0f5ae47fd51b2418b1355a2e819e58b69890/text-generation-inference/server/pyproject.toml#L17

File: optimum/neuron/trainers.py
Comment (michaelbenayoun) [reply optimum/neuron/trainers.py:1377]: Fixed, it was a quick test artifact

File: tests/test_trainers.py
Comment (michaelbenayoun) [reply tests/test_trainers.py]: Done!

**Review by dacorvo — APPROVED** [review]
LGTM, thanks ! If you can fix the seq2seq tracing issue before merging that pull-request that is even better.

**Comment by michaelbenayoun** [comment]
I fixed all but one test.

**Pull request merged**
